sktime/forecasting/timesfm_forecaster.py

```diff
@@ -199,30 +200,44 @@ def __init__(
         super().__init__()
 
     def _fit(self, y, X, fh):
-        # import after backend env has been set
-        if self.use_source_package:
-            from timesfm import TimesFm
-        else:
-            from sktime.libs.timesfm import TimesFm
-
         if fh is not None:
             fh = fh.to_relative(self.cutoff)
             self._horizon_len = max(self.horizon_len, *fh._values.values)
         else:
             self._horizon_len = self.horizon_len
 
-        self.tfm = TimesFm(
-            context_len=self.context_len,
-            horizon_len=self._horizon_len,
-            input_patch_len=self.input_patch_len,
-            output_patch_len=self.output_patch_len,
-            num_layers=self.num_layers,
-            model_dims=self.model_dims,
-            per_core_batch_size=self.per_core_batch_size,
-            backend=self.backend,
-            verbose=self.verbose,
-        )
-        self.tfm.load_from_checkpoint(repo_id=self.repo_id)
+        self.tfm = _CachedTimesFM(
+            key=self._get_unique_timesfm_key(),
+            timesfm_kwargs=self._get_timesfm_kwargs(),
+            use_source_package=self.use_source_package,
+            repo_id=self.repo_id,
+        ).load_from_checkpoint()
+
+    def _get_timesfm_kwargs(self):
+        """Get the kwargs for TimesFM model."""
+        return {
+            "context_len": self.context_len,
+            "horizon_len": self._horizon_len,
+            "input_patch_len": self.input_patch_len,
+            "output_patch_len": self.output_patch_len,
+            "num_layers": self.num_layers,
+            "model_dims": self.model_dims,
+            "per_core_batch_size": self.per_core_batch_size,
+            "backend": self.backend,
+            "verbose": self.verbose,
+        }
+
+    def _get_unique_timesfm_key(self):
+        """Get unique key for TimesFM model to use in multiton."""
+        repo_id = self.repo_id
+        use_source_package = self.use_source_package
+        kwargs = self._get_timesfm_kwargs()
+        kwargs_plus_repo_id = {
```

Comment:
Order-dependence of Python dictionaries is really a byproduct of current implementation. I think it would be safer to use `str(sorted(kwargs_plus_repo_id.items()))`

Reply:
well, worst case it will just spawn more copies than necessary, but I get your point, thanks for the suggestion.

Reply:
thanks, implemented as suggested!